puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -26,7 +26,18 @@ def save(request)
         facts = request.instance.dup
         facts.values = facts.values.dup
         facts.values[:trusted] = get_trusted_info(request.node)
-        {
+
+        inventory = facts.values['_puppet_inventory_1']
+        package_inventory =
+          if inventory and inventory.respond_to?(:keys) and inventory['packages']
+            inventory['packages'].map { |p|
+              [p['package_name'], p['version'], p['provider']]
```

Comment:
if we could do that we could also split out the array-valued versions in the fact definition and not have to worry about any of that in the terminus or application